File: src/main/java/cz/encircled/eplayer/service/MediaService.java
package cz.encircled.eplayer.service;

import cz.encircled.eplayer.model.MediaType;
import org.jetbrains.annotations.NotNull;

/**
 * @author Encircled on 9.6.2014.
 */
public interface MediaService {

    void releasePlayer();

    void updateCurrentMediaInCache();

    void play(@NotNull String path);

    /**
     * @param p
     */
    void play(@NotNull MediaType p);

    long getMediaLength();

    long getCurrentTime();

    int getVolume();

    void setVolume(int value);

    void setTime(long value);

    void start();

    void toggle();

    void pause();

    void stop();

    int getSubtitles();

    void setSubtitles(int id);

    int getAudioTrack();

    void setAudioTrack(int trackID);

    boolean isPlaying();

}

File: src/main/java/cz/encircled/eplayer/service/MediaService.kt
package cz.encircled.eplayer.service

import cz.encircled.eplayer.model.GenericTrackDescription
import cz.encircled.eplayer.model.PlayableMedia
import uk.co.caprica.vlcj.player.component.EmbeddedMediaPlayerComponent

/**
 * @author Encircled on 9.6.2014.
 */
interface MediaService {

    var volume: Int

    fun currentSubtitle(): GenericTrackDescription?

    fun setSubtitle(track: GenericTrackDescription, byUser: Boolean = true)

    fun currentAudioTrack(): GenericTrackDescription?

    fun setAudioTrack(track: GenericTrackDescription, byUser: Boolean = true)

    fun currentMedia(): PlayableMedia?

    fun createPlayer(): EmbeddedMediaPlayerComponent

    fun releasePlayer()

    fun isPlaying(): Boolean

    fun play(path: String)

    fun play(media: PlayableMedia)

    fun playNext()

    fun playPrevious()

    fun setTime(value: Long)

    /**
     * Add $value ms to the current time
     */
    fun setTimePlus(value: Long)

    fun start()

    fun toggle()

    fun pause()

    fun stop()

    fun switchAudioPassThruIfNeeded(track: GenericTrackDescription)
}